extras/R6ClassFun.R

```diff
@@ -357,13 +355,14 @@ plpModuleSpecifications <- plpModuleSettingsCreator$createModuleSpecifications(
 analysisSpecifications <- createEmptyAnalysisSpecificiations() |>
   addSharedResources(cohortSharedResourcesSpecifications) |>
   addSharedResources(ncoCohortSharedResourceSpecifications) |>
+  # WORKING
   addModuleSpecifications(cgModuleSettings) |>
   addModuleSpecifications(cdModuleSpecifications) |>
   addModuleSpecifications(cmModuleSpecifications) |>
-  addModuleSpecifications(sccsModuleSpecifications)
-  # NOT WORKING
+  addModuleSpecifications(sccsModuleSpecifications) |>
```

Comment:
It would be really nice if this all was more context specific. `addModuleSpecifications` seems like it's just a generic function that can accept anything. Wouldn't the above look better as:

```
  addCohortSpecifications(cohortSharedResourcesSpecifications) |>
  addNegativeControls(ncoCohortSharedResourceSpecifications) |>
  addCohortGeneratorSpecs(cgModuleSettings) |>
  addCohortDiagnostics(cdModuleSpecifications) |>
  addCohortMethod(cmModuleSpecifications) |>
  addSelfControlCaseSeries(sccsModuleSpecifications)
```

Reply:
Agreed - it would be nicer to have a more explicit set of functions like you are showing here. I'll link this comment to #144 since this would support the notion of explicitly named items in the analysis specification.

Reply:
From discussion w/ @chrisknoll I'll add these functions as a way to ensure the module settings are validated when assembling the analysis specification.

Reply:
@chrisknoll check out https://github.com/OHDSI/Strategus/pull/145/commits/758c0d3d6dac33aefe1695b799a6151293dd2ad4. I stopped short of the "sharedResources" settings since those will probably go away per #144 but like where this is headed as now each of these functions can use the R6 class to validate the settings.